extensions/elytron-security-oauth2/deployment/src/main/java/io/quarkus/elytron/security/oauth2/deployment/OAuth2DeploymentProcessor.java

```diff
@@ -0,0 +1,118 @@
+package io.quarkus.elytron.security.oauth2.deployment;
+
+import org.jboss.logging.Logger;
+import org.wildfly.security.auth.server.SecurityRealm;
+
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.substrate.SubstrateResourceBuildItem;
+import io.quarkus.elytron.security.deployment.*;
+import io.quarkus.elytron.security.oauth2.runtime.OAuth2Config;
+import io.quarkus.elytron.security.oauth2.runtime.OAuth2Recorder;
+import io.quarkus.elytron.security.runtime.*;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.undertow.deployment.ServletExtensionBuildItem;
+import io.undertow.security.idm.IdentityManager;
+import io.undertow.servlet.ServletExtension;
+
+/**
+ * The build time process for the OAUth2 security aspects of the deployment. This creates {@linkplain BuildStep}s for
+ * integration
+ * with the Elytron OAUth2 security services. This supports the Elytron OAuth2
+ * {@linkplain org.wildfly.security.auth.realm.token.TokenSecurityRealm realm implementations.
+ */
+class OAuth2DeploymentProcessor {
+    private static final Logger log = Logger.getLogger(OAuth2DeploymentProcessor.class.getName());
+    private static final String REALM_NAME = "Oauth2";
+    private static final String AUTH_MECHANISM = "BEARER_TOKEN";
+
+    OAuth2Config oauth2;
+
+    /**
+     * If the configuration specified a deployment local key resource, register it with substrate
+     *
+     * @return SubstrateResourceBuildItem
+     */
+    @BuildStep
+    SubstrateResourceBuildItem registerSubstrateResources() {
+        if (oauth2.caCertFile.isPresent()) {
+            String publicKeyLocation = oauth2.caCertFile.get();
+            if (publicKeyLocation.indexOf(':') < 0 || publicKeyLocation.startsWith("classpath:")) {
+                log.infof("Adding %s to native image", publicKeyLocation);
+                return new SubstrateResourceBuildItem(publicKeyLocation);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Register this extension as a security-oauth2feature
+     *
+     * @return
+     */
```

Comment:
We really don't need comments when things are obvious. It's counter productive.

Reply:
done